re-save blobs when saving error forms

snopoke opened the pull request:
##### SUMMARY
When an error happens after the form is saved the transaction is rolled back. This has some effects:
* form is not saved to db but `form.pk` is not cleared
* form attachments are written to the blob db but the blobdb meta is not saved
* although the blob meta isn't saved their 'saved' state is kept on the form object so when saving the error form the blobs are not re-saved

This PR addresses those by:
1. clear `model.pk` if there is a DB error
2. Separate committing of saved attachment state from the actual saving and only commit if the transactions was successful (using `transaction.on_commit`

ShubhamBansal1997 committed:
[build]: Migration Form Model
- const.py -> added the constants for the table
- aggreate.py -> model for migration table
- migration_form.py -> migration helper

- JIRA Ticket:
    https://dimagi-dev.atlassian.net/browse/ICDS-1025
- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor linting fixes

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: include the MigrationFormsAggregationDistributedHelper in the
__init__.py file and correct the unique_together in the django model definition

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[migration]: added the migratio file

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: remove extra autogenerated migration code

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor fixes
- removed extra field from the migration model last_time_processed
- taking previous month data into account while getting data from the
UCR

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: removed extra field

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[refactor]: updated the migration_status in the CCSRecordMonthlyHelper
from the person_cases to migration_table

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[build]: Updated the aggregation output due to addition of the
miigration form

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: resovle the migration merge issues

django.core.management.base.CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0158_auto_20191227_1147, 0159_update_agg_awc_monthly_view_visit_fields in icds_reports).

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[refactor]: change the type of the migration_status to postive int in
-> ucr
-> model
-> migration

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: made the AggregateMigrationForms distributed

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: fix a failing test custom.icds_reports.tests.agg_tests.reports.test_demographics:TestDemographics.test_data_daily_if_aggregation_script_fail_lactating_enrolled_women

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor name fix agg_min -> agg_migration
migration_status default to 0 instead of null

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[revert]: reverting back the adding of the migration model in the ccs
record monthy will do after merging of this PR

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[revert]: since the UCR is migration, so values are already stored as
string, which are difficult to migrate

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[refactor]: shifting migrations_form.json to dashboard since it's being
used in the dashboard

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[refactor]: converting migration_status into 1, 0 forms at the agg level

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[revert]: reverting test changes

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[revert]: revert back the ccs record sql

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: fix the test_migration_form

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: added new column is_migrated and using the same for handling the
migration_status

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor fixes
- changing the name of the field migration_status to is_migrated

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor column name fix

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[chore]: adding the agg migration form tasks to airflow

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

orangejenny committed:
Added SQLRegistrationRequest (PR1)

orangejenny committed:
Added populate_sql_registration_request (PR1)

orangejenny committed:
Added save to SQL (PR1)

orangejenny committed:
Added django migration to enforce data migration (PR2)

orangejenny committed:
Updated reads to read from SQL (PR2)

orangejenny committed:
Added RegistrationRequest to deletable_doc_types (PR2)

Rohit25negi committed:
updated the aggregation for cbe and vhnd count

Rohit25negi committed:
fixed test cases

Rohit25negi committed:
updated aggregation output

Rohit25negi committed:
Removed unused imports

challabeehyv committed:
removed feature flag

ShubhamBansal1997 committed:
[fix]: added the migration date for migration aggregation migration

- Github Issue:

ShubhamBansal1997 committed:
[fix]: minor fixes

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

challabeehyv committed:
removed feature flag. merged adolscent girl data to original data and removed dailt tests

ShubhamBansal1997 committed:
[fix]: minor test fixes and tasks fixes

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor typo fix

- Github Issue:

ShubhamBansal1997 committed:
[refactor]: removed too much migration file and transformed into one

- Github Issue:

ShubhamBansal1997 committed:
[fix]: remove the extra option

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: minor typo fix

- Github Issue:

ShubhamBansal1997 committed:
[revert]: changes in the model

- Github Issue:

ShubhamBansal1997 committed:
[fix]: minor fix in sql query

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[refactor]: added more tests

- Github Issue:

challabeehyv committed:
changed the cast value from integer to text

orangejenny committed:
Fixed whitespace

orangejenny committed:
Added unique contraint and wrap call

orangejenny committed:
Added domain deletion handling

orangejenny committed:
Added SQLDefaultConsumption (PR1)

orangejenny committed:
Added populate_defaultconsumption (PR1)

orangejenny committed:
Added cleanup when deleting domain (PR1)

orangejenny committed:
Added sync code (PR1)

challabeehyv committed:
Merge pull request #64 from dimagi/master

updating fork

orangejenny committed:
Merge branch 'master' into jls/migrate-RegistrationRequest-1

orangejenny committed:
Removed couch_key

orangejenny committed:
Fixed class name in DOMAIN_DELETE_OPERATIONS

orangejenny committed:
Merge branch 'master' into jls/migrate-DefaultConsumption-1

orangejenny committed:
Removed couch_key

orangejenny committed:
Switched migration to use update_or_create

See https://github.com/dimagi/commcare-hq/pull/26617

orangejenny committed:
Added choices to SQLDefaultConsumption.type

orangejenny committed:
Removed unused import

biyeun committed:
send errors to datadog on the process bounced emails task

biyeun committed:
add datadog counters for all the different error types in bounced_email_manager

orangejenny committed:
Removed wrapping in SQLRegistrationRequest.save

saiprakash-v committed:
Merge pull request #20 from dimagi/master

updating master

Rohit25negi committed:
Merge branch 'master' into rn_add_cbe_vhnd_for_sdr

Rohit25negi committed:
Fixed migration conflict

challabeehyv committed:
changed the aggregation level to original value

orangejenny committed:
Merge branch 'jls/migrate-RegistrationRequest-1' into jls/migrate-RegistrationRequest-2

orangejenny committed:
Added datetime conversions

orangejenny committed:
Merge branch 'jls/migrate-RegistrationRequest-1' into jls/migrate-RegistrationRequest-2

orangejenny committed:
Fixed always creating new doc on sql save

orangejenny committed:
Switched to use sync mixins

orangejenny committed:
Merge branch 'jls/migrate-RegistrationRequest-1' into jls/migrate-RegistrationRequest-2

orangejenny committed:
Removed unused imports

orangejenny committed:
Merge branch 'jls/migrate-RegistrationRequest-1' into jls/migrate-RegistrationRequest-2

dannyroberts committed:
Merge branch 'master' into dmr/test-es-2.4.6

saiprakash-v committed:
Adding additional filters for mobile dashboard

saiprakash-v committed:
adding additional filter directive in shared components

saiprakash-v committed:
tests for additional filters

saiprakash-v committed:
fixing lint errors

czue committed:
fail with clearer error if shape is missing from file

czue committed:
add new J&K and Ladakh district files, and make clickable again

challabeehyv committed:
Merge pull request #65 from dimagi/master

updating fork

saiprakash-v committed:
removing shared directive from dashboard.html and mocha.html

biyeun committed:
prepend datadog_counter tags with commcare.util.

orangejenny committed:
Switched to LocationsSearchView for mobile users download

orangejenny committed:
Switched to LocationsSearchView for edit mobile worker page

biyeun committed:
bring back the try except so i can get more detailed information if this errors out

orangejenny committed:
Added include_suffixes option to strip '[location]' on location-only search options

orangejenny committed:
Switched default query for LocationSelectWidget to use location_search

orangejenny committed:
Removed child_locations_for_select2

esoergel committed:
Work with text, not bytestrings

This was failing elsewhere where the write wasn't being encoded back to
bytes

orangejenny committed:
Consolidated help text

esoergel committed:
Use StringIO also in tests

challabeehyv committed:
Merge pull request #66 from dimagi/master

updating fork

Rohit25negi committed:
Take the launched numbers from the agg_Awc table only for the current date

Rohit25negi committed:
Fix test cases

challabeehyv committed:
changed nu lssl aunched text to Not launched instead of data not enetered

saiprakash-v committed:
using provide default constants in tests

Rohit25negi committed:
Added SDR eleent to Tabular report UNDER FF

Rohit25negi committed:
Fix UI test cases

Rohit25negi committed:
Accept request for the tabular report

saiprakash-v committed:
refactoring and code comments

saiprakash-v committed:
lint error and utils declaration

challabeehyv committed:
added the check to add ls indicator to reports from jan 2020

Rohit25negi committed:
Improved the code style

snopoke committed:
test missing XML when handling error

snopoke committed:
test missing XML when handling error

snopoke committed:
clear primary keys on db error

snopoke committed:
clear primary keys on db error

snopoke committed:
separate write and commit for form attachments

In the case of an error the attachmets would be deleted form the blob db
but the list of attachments on the form would still reference the MetaDB
objects and so when saving the error the attachments would be lost.

snopoke committed:
separate write and commit for form attachments

In the case of an error the attachmets would be deleted form the blob db
but the list of attachments on the form would still reference the MetaDB
objects and so when saving the error the attachments would be lost.

snopoke committed:
use more generic error

snopoke committed:
use more generic error

snopoke committed:
drop nested transaction

snopoke committed:
drop nested transaction

snopoke committed:
add test case for error during edit

snopoke committed:
add test case for error during edit

Rohit25negi committed:
improved readability more

orangejenny committed:
Added check for element before applying bindings

Rohit25negi committed:
kwarg

orangejenny committed:
Fixed test_delete_domain

orangejenny committed:
Merge pull request #26654 from dimagi/jls/check-for-el-before-applying-bindings

Added check for element before applying bindings

orangejenny committed:
Added required fields in test_delete_domain

orangejenny committed:
Removed zipline directory

orangejenny committed:
Removed ilsgateway directory

orangejenny committed:
Added ILSGatewayConfig to deletable doc types

orangejenny committed:
Removed urls

orangejenny committed:
Removed toggle

This was only referenced in custom/ilsgateway

orangejenny committed:
Removed ilsgateway custom queue reference from celery docs

orangejenny committed:
Removed CUSTOM_SMS_HANDLERS and CUSTOM_PROJECT_SMS_QUEUES

orangejenny committed:
Removed CUSTOM_DASHBOARD_PAGE_URL_NAMES

orangejenny committed:
Removed remaining references in settings.py

esoergel committed:
Merge pull request #26648 from dimagi/es/dump-reload

Use string encoding in dump/reload code

biyeun committed:
remove .util

orangejenny committed:
Updated test_delete_domain so activatioon_guiid is unique

millerdev reviewed (approved):
Form and case saving is quite an intricate process, so the new tests are a great addition. Unfortunately I think this has caused other test failures (example: `corehq.apps.couch_sql_migration.tests.test_casedifftool:TestCouchSqlDiff.test_couch_with_missing_forms`).

snopoke committed:
remove tzinfo

orangejenny committed:
Merge pull request #26555 from dimagi/jls/migrate-RegistrationRequest-1

Migrate RegistrationRequest to SQL (1 of 4)

orangejenny committed:
Merge pull request #26602 from dimagi/jls/migrate-DefaultConsumption-1

Migration DefaultConsumption to SQL (1 of 3)

biyeun committed:
add additional datadog counters for success tracking

biyeun committed:
add feature flag for allowing De-ID in Odata feeds

biyeun committed:
implement feature flag in create odata feed

biyeun committed:
Merge pull request #26621 from dimagi/bmb/SAASP-10229

[AWS SES] Add datadog counters and error tracking to bounced email manager

orangejenny committed:
Fixed error in DefaultConsumption migration when consumption is None

orangejenny committed:
Altered request_ip to allow nulls

orangejenny committed:
Merge pull request #26659 from dimagi/jls/alloow-blank-consumption

Fixed error in DefaultConsumption migration when consumption is None

orangejenny committed:
Merge pull request #26660 from dimagi/jls/allow-blank-request-ip

Altered request_ip to allow nulls

orangejenny committed:
Updated blank/None logic in default consumption migration

orangejenny committed:
Merge pull request #26661 from dimagi/jls/blank-is-not-none

Updated blank/None logic in default consumption migration

orangejenny committed:
Renamed variable

orangejenny committed:
Moved decorator from dispatch method to class

orangejenny committed:
Renamed variable

calellowitz committed:
Merge pull request #26649 from dimagi/rn_take_launched_from_agg_awc

Take the launched numbers from the agg_Awc table only for the current date

dannyroberts committed:
Merge branch 'master' into dmr/test-es-2.4.6

dannyroberts committed:
Increment ELASTICSEARCH_MAJOR_VERSION in localsettings files, not settings.py for now

czue committed:
Merge pull request #26635 from dimagi/cz/j-k-2

add new J&K and Ladakh district files, and make clickable again

Rohit25negi committed:
Merge pull request #26527 from dimagi/rn_add_cbe_vhnd_for_sdr

[Icds Dashboard] Rn add cbe vhnd for Service Delivery Report

saiprakash-v committed:
explicitly initializing showGender and age to false

snopoke committed:
Update custom/icds/tasks/__init__.py

snopoke committed:
fix noop case

snopoke committed:
fix noop case

challabeehyv committed:
test case added

czue committed:
remove openstreetmap option from dashboard

challabeehyv committed:
added a TODO

snopoke committed:
Merge pull request #26657 from dimagi/sk/date-compare

remove tzinfo

snopoke committed:
exclude cases if None

snopoke committed:
exclude cases if None

challabeehyv committed:
removed the unused code to remove date from config

orangejenny committed:
Merge pull request #26656 from dimagi/jls/kill-ilsgateway-and-zipline

Removed ilsgateway and zipline

czue committed:
Merge pull request #26662 from dimagi/cz/mapbox

ICDS Dashboard: remove openstreetmap option from dashboard

biyeun committed:
Merge pull request #26658 from dimagi/bmb/SAASP-10270

Add Feature Flag for Using De-Identification in OData Feeds

dannyroberts committed:
Merge pull request #25898 from dimagi/dmr/test-es-2.4.6

Use elasticsearch 2.4.6 in docker setup

ShubhamBansal1997 committed:
[refactor]: remove extra column

- Github Issue:

ShubhamBansal1997 committed:
[fix]: fix test as per the change in the migration ucr

- Github Issue:

ShubhamBansal1997 committed:
[revert]: add the param removed accidently

- Github Issue:

czue committed:
Merge pull request #26634 from saiprakash-v/additional-filters

ICDS Dashboard: Additional filters for mobile dashboard

ShubhamBansal1997 committed:
Merge branch 'master' into sb/migration_model

Rohit25negi committed:
Merge pull request #26653 from challabeehyv/cr/qa-1022

[ICDS Dashboard] Made the LS indicator field available in system usage report from jan 2020

Rohit25negi committed:
Merge pull request #26626 from challabeehyv/cr/icds-1129

[ICDS Dashboard] Changed the incremented aggregation level to original value for undernutrition data

Rohit25negi committed:
Merge pull request #26543 from challabeehyv/cr/tabular_activity_report

[ICDS Dashboard] Chaning the indicator casting value from integer to text for tabular data pull

Rohit25negi committed:
Merge pull request #26532 from challabeehyv/cr/icds-905

[ICDS Dashboard] Removed the feature flag for ICDS-904

Rohit25negi committed:
Merge pull request #26650 from challabeehyv/cr/qa-1023

[ICDS Dashboard] Fixed the incorrect message displayed when no LS is launched for a location

kaapstorm committed:
Refactor: Property to param

kaapstorm committed:
Refactor: Extract function

mkangia committed:
share only usernames

mkangia committed:
include location name in filename

kaapstorm committed:
Fix: Append values

kaapstorm committed:
Fix: Skip special value

kaapstorm committed:
Tests

orangejenny committed:
Added a few real examples to style guide [ci skip]

ShubhamBansal1997 committed:
[fix]: migration fix

- Github Issue:

orangejenny committed:
Merge branch 'master' into jls/migrate-RegistrationRequest-2

orangejenny committed:
Reverted "Added RegistrationRequest to deletable_doc_types (PR2)"

This reverts commit a946a89edaa59a93dafe5e0b3cf9f247741e53ff.

orangejenny committed:
Update staging.yaml

orangejenny committed:
Incremented migration

ShubhamBansal1997 committed:
Merge pull request #26268 from dimagi/sb/migration_model

[ICDS Dashboard]Added the Migration Form Model

orangejenny committed:
Merge pull request #26556 from dimagi/jls/migrate-RegistrationRequest-2

Migrate RegistrationRequest to SQL (2 of 4)

esoergel committed:
Don't include billing context on enterprise envs

esoergel committed:
Include trace in error

This shouldn't really be used on ENTERPRISE_MODE domains.  This trace
will help us figure out where the call is coming from.

orangejenny committed:
Fixed search param for locations widget

orangejenny committed:
Fixed search param and processResults for create new mobile worker

orangejenny committed:
Removed unused LocationUserForm

orangejenny committed:
Updated create new mobile worker to use standard locations widget (ignore whitespace)

orangejenny committed:
Removed unused param

kaapstorm committed:
Merge pull request #26666 from dimagi/nh/omrs/missing_values

Nh/omrs/missing values

mkangia committed:
correct naming

orangejenny committed:
Merge pull request #26646 from dimagi/jls/reconcile-location-searches

Reconcile location searches

esoergel committed:
Merge pull request #26669 from dimagi/es/enterprise-billing

Don't include billing context in ENTERPRISE_MODE

dannyroberts committed:
Add management command delete_form_attachments

saiprakash-v committed:
Merge pull request #21 from dimagi/master

updating master

saiprakash-v committed:
Optimizing how maps look like on different screens

mkangia committed:
Merge pull request #26665 from dimagi/mk/dump-usernames-followup

Dump usernames followup

mkangia committed:
Merge pull request #26667 from dimagi/jls/styleguide-examples

Added a few real examples to style guide [ci skip]

saiprakash-v committed:
changing additional filters heading

snopoke committed:
raise ESError when unable to fetch all docs

snopoke committed:
catch correct error

snopoke committed:
catch correct error

snopoke committed:
stickler

snopoke committed:
stickler

Rohit25negi committed:
better message
used global variable

Rohit25negi committed:
Satisfy lynter

czue committed:
Merge pull request #26675 from saiprakash-v/additional-filters-heading

ICDS Dashboard: Changing additional filters heading

challabeehyv committed:
reverting the old change which is causing the ranks to not load

czue committed:
Merge pull request #26674 from saiprakash-v/maps-ui-optimization

ICDS Dashboard: Making maps responsive for different screen sizes

Rohit25negi committed:
Merge pull request #26652 from dimagi/rn_add_ui_element_tabular_report

[ICDS Dashboard] Add ui element tabular report

orangejenny committed:
Update expired_domains.md

orangejenny committed:
Added string casts for usernames

orangejenny committed:
Added accounting for None

orangejenny committed:
Merge pull request #26680 from dimagi/orangejenny-patch-2

Update expired_domains.md

orangejenny committed:
Fixed locations not saving on various pages

orangejenny committed:
Merge pull request #26682 from dimagi/jls/emwf-location-ids

Fixed locations not saving on various pages

orangejenny committed:
Updated location_tree to not look for l__ prefix

orangejenny committed:
Merge pull request #26683 from dimagi/jls/emwf-main-loc-search

Updated location_tree to not look for l__ prefix

orangejenny committed:
Removed now-unnecessary computed observable

Rohit25negi committed:
Merge pull request #26678 from challabeehyv/cr/icds-1129

[ICDS Dashboard] reverting the old change which is causing the ranks to not load

orangejenny committed:
Merge pull request #26681 from dimagi/jls/cast-usernames-in-bulk-delete

Added string casts for usernames

orangejenny committed:
Update staging.yaml

millerdev commented on corehq/form_processor/backends/sql/processor.py:
```diff
@@ -104,7 +104,11 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None, p
+        all_models = filter(None, (
+            list(processed_forms) +
+            (cases if cases else []) +
+            (stock_result.models_to_save if stock_result else [])
```
nit: this is a perfect place for `itertools.chain` (to avoid list concatenation)
```py
all_models = filter(None, chain(
    processed_forms,
    cases or [],
    stock_result.models_to_save if stock_result else [],
))
```

millerdev commented on corehq/form_processor/backends/sql/processor.py:
```diff
@@ -104,30 +104,42 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None, p
+            list(processed_forms)
+            + (cases if cases else [])
+            + (stock_result.models_to_save if stock_result else [])
+        ))
```
nit: this is a perfect place for `itertools.chain` (to avoid constructing a list that is not used except in case of error)
```py
all_models = filter(None, chain(
    processed_forms,
    cases or [],
    stock_result.models_to_save if stock_result else [],
))
```

dannyroberts committed:
Add link to suggestion in PR discussion

dannyroberts committed:
Merge pull request #26672 from dimagi/dmr/delete_form_attachments

Add management command delete_form_attachments

orangejenny committed:
Merge pull request #26684 from dimagi/jls/location-search-followup

Removed now-unnecessary computed observable

snopoke committed:
inline import

snopoke committed:
Merge pull request #26677 from dimagi/sk/catch-es-error

raise ESError when unable to fetch all docs

snopoke committed:
use itertools

snopoke committed:
refetch form in tests

This is necessary in tests since the post commit actions
on SQL transactions are never excercised because the test
classes run each test in its own transaction which gets rolled
back at the end of the test.

snopoke committed:
tear down

snopoke committed:
Merge branch 'sk/errors' of github.com:dimagi/commcare-hq into sk/errors

snopoke commented:
history got messed up, closing this in favour of https://github.com/dimagi/commcare-hq/pull/26693
